src/CMakeLists.txt

```diff
@@ -0,0 +1,79 @@
+add_subdirectory(vmec_interface)
+
+
+add_executable(stella.x  ${SOURCES} )
+
+#set flags here
+#TODO add the most common flags needed for stella
+
+set(dialect "-ffree-line-length-none -fno-backslash -I /usr/include")
+set(bounds "-fbounds-check")
+set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} ${dialect} -march=native")
+set(CMAKE_Fortran_FLAGS_DEBUG "${CMAKE_Fortran_FLAGS_DEBUG} ${bounds} -pedantic -g -Wall -fimplicit-none")
+if(PRECISION STREQUAL "DOUBLE")
+	set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fdefault-real-8 -fdefault-double-8")
+endif()
+
+#get package includes and libs here
+
+find_package(MPI REQUIRED)
+if(MPI_FOUND)
+	message(STATUS "Found MPI Libraries in ${MPI_LIBRARIES}")
+	get_filename_component (MPI_LIBRARY_PATH ${MPI_LIBRARY} DIRECTORY)
+    target_include_directories(stella.x PRIVATE ${MPI_Fortran_INCLUDE_PATH})
+	target_link_libraries( stella.x PRIVATE ${MPI_Fortran_LIBRARIES})
+endif()
+
+find_package(LAPACK REQUIRED)
+if(LAPACK_FOUND)
+	message(STATUS "Found LAPACK Libraries in ${LAPACK_LIBRARIES}")
+	target_link_libraries( stella.x PRIVATE ${LAPACK_LIBRARIES})
+endif()
+
+
+find_package(FFTW REQUIRED)
+if(FFTW_FOUND)
+	message(STATUS "Found FFTW Libraries in ${FFTW_LIBRARIES}")
+	include_directories(${FFTW_INCLUDES})
+	target_link_libraries(stella.x PRIVATE ${FFTW_LIBRARIES})
+endif()
+
+
+find_package(PETSc REQUIRED)
+if(PETSc_FOUND)
+	message(STATUS "Found PETSc Libraries in ${PETSC_LIBRARIES}")
+	include_directories(${PETSC_INCLUDES})
+	target_link_libraries(stella.x PRIVATE ${PETSC_LIBRARIES})
+endif()
+
+#link executable to internal library
+target_include_directories(stella.x PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/vmec_interface/)
+target_link_libraries(stella.x PRIVATE mini_libstell)
+
+
+#TODO configure FindNetCDF
+set (NETCDF_F90 "YES")
+find_package(NetCDF REQUIRED)
+
+target_include_directories(stella.x PRIVATE "/opt/netcdf-fortran-4.5.3/include")
+target_link_libraries(stella.x PRIVATE "/opt/netcdf-fortran-4.5.3/lib/libnetcdff.so")
+
+
+#get all fpp and f90 files
+file(GLOB SOURCES *.f90 utils/*.f90 geo/*.f90)
+file(GLOB SOURCES_FPP *.fpp utils/*.fpp )
```

Comment:
The CMake developers recommend against using `GLOB` -- the issue is that it makes it harder to work out which files need to be rebuilt, as it has to search for all possible files each time you re-run cmake.

It's better to be explicit and list all of the files, even if it's a bit more painful